__tests__/validations/actions.test.tsx

```diff
@@ -0,0 +1,549 @@
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  validatePrefabs([prefab]);
+
+  t.pass();
+});
+
+test('Throw when action has no name', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [{} as PrefabAction],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  t.throws(() => validatePrefabs([prefab]), {
+    message: `
+Property: "actions[0].name" is required at prefab: Prefab
+`,
+  });
+});
+
+test('Throw when action has no useNewRuntime', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [{ name: 'foo' } as PrefabAction],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  t.throws(() => validatePrefabs([prefab]), {
+    message: `
+Property: "actions[0].useNewRuntime" is required at prefab: Prefab
+`,
+  });
+});
+
+test('Throw when useNewRuntime is not a boolean', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [
+      ({
+        name: 'foo',
+        useNewRuntime: '1',
+      } as unknown) as PrefabAction,
+    ],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  t.throws(() => validatePrefabs([prefab]), {
+    message: `
+Property: "actions[0].useNewRuntime" must be a boolean at prefab: Prefab
+`,
+  });
+});
+
+test('Throw when action has no ref', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [{ name: 'foo', useNewRuntime: true } as PrefabAction],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  t.throws(() => validatePrefabs([prefab]), {
+    message: `
+Property: "actions[0].ref" is required at prefab: Prefab
+`,
+  });
+});
+
+test('Throw when action has no id inside ref', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [
+      {
+        name: 'foo',
+        useNewRuntime: true,
+        ref: { endpointId: 'bar' },
+      } as PrefabAction,
+    ],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  t.throws(() => validatePrefabs([prefab]), {
+    message: `
+Property: "actions[0].ref.id" is required at prefab: Prefab
+`,
+  });
+});
+
+test('Throw when action has no endpointId inside ref', (t: Context): void => {
+  const prefab = {
+    category: 'CONTENT',
+    icon: 'TitleIcon',
+    actions: [
+      {
+        name: 'foo',
+        useNewRuntime: true,
+        ref: { id: 'bar' },
+      } as PrefabAction,
+    ],
+    name: 'Prefab',
+    structure: [],
+  } as Prefab;
+
+  t.throws(() => validatePrefabs([prefab]), {
+    message: `
+Property: "actions[0].ref.endpointId" is required at prefab: Prefab
+`,
+  });
+});
+
+test('Pass when actions contains an event of a kind supported by the old runtime', (t: Context): void => {
```

Comment:
Should a user be able to have an action in his/her prefab without using the `ref` in the components?

Reply:
Yes, you can't have an action entity without a ref. Why create an action that is connected to nothing.